notificationdemo/source/android/app/src/main/java/org/kaaproject/kaa/demo/notification/MainActivity.java

```diff
@@ -0,0 +1,218 @@
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.util.Log;
+import android.view.View;
+
+import org.kaaproject.kaa.client.notification.NotificationListener;
+import org.kaaproject.kaa.client.notification.NotificationTopicListListener;
+import org.kaaproject.kaa.common.endpoint.gen.Topic;
+import org.kaaproject.kaa.demo.notification.entity.TopicPojo;
+import org.kaaproject.kaa.demo.notification.fragment.NotificationDialogFragment;
+import org.kaaproject.kaa.demo.notification.fragment.NotificationFragment;
+import org.kaaproject.kaa.demo.notification.fragment.OnFragmentUpdateEvent;
+import org.kaaproject.kaa.demo.notification.fragment.TopicFragment;
+import org.kaaproject.kaa.demo.notification.kaa.KaaManager;
+import org.kaaproject.kaa.demo.notification.storage.TopicStorage;
+import org.kaaproject.kaa.demo.notification.util.NotificationConstants;
+import org.kaaproject.kaa.demo.notification.util.TopicHelper;
+import org.kaaproject.kaa.schema.sample.notification.SecurityAlert;
+
+import java.util.List;
+
+import static org.kaaproject.kaa.demo.notification.util.TopicHelper.addNotificationToTopic;
+
+/**
+ * The implementation of the {@link FragmentActivity} class.
+ * Use multithreading for starting Kaa client
+ * Init Notification and Topic listener, which can get information from server
+ * Manage all views.
+ * <p>
+ * Tip: you can use Service with this AsyncTask and send all Kaa Notification in mobile notification area
+ */
+public class MainActivity extends AppCompatActivity implements TopicFragment.OnTopicClickedListener {
+
+    private KaaManager manager;
+    private NotificationListener notificationListener;
+    private NotificationTopicListListener topicListener;
+
+    @Override
+    public void onCreate(Bundle savedInstance) {
+        super.onCreate(savedInstance);
+        setContentView(R.layout.activity_main);
+
+        setUpToolbar();
+        showProgress();
+        startKaa();
+    }
+
+    private void setUpToolbar() {
+        setSupportActionBar((Toolbar) findViewById(R.id.toolbar));
+    }
+
+    public void startKaa() {
+        kaaTask.execute();
+    }
+
+    // Tip: you can use it from Service
+    private AsyncTask<Void, Void, Void> kaaTask = new AsyncTask<Void, Void, Void>() {
+        @Override
+        protected Void doInBackground(Void... params) {
+            initNotificationListener();
+            initTopicListener();
+
+            manager = new KaaManager(MainActivity.this);
+            manager.start(notificationListener, topicListener);
+
+            syncTopics();
+
+            return null;
+        }
+
+        @Override
+        protected void onPostExecute(Void aVoid) {
+            super.onPostExecute(aVoid);
+
+            hideProgress();
+            addTopicsFragment();
+        }
+    };
+
+    private void syncTopics() {
+        List<TopicPojo> syncedTopics = TopicHelper.syncTopics(TopicStorage.get().getTopics(), manager.getTopics());
```

Comment:
@maxml Agree, I simplified TopicStorage to use only LinkedHashList without shared preferences